File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size()==0) {
    return 0;
}
map<int, int> hash;
for (auto i : nums) {
    hash[i]++;
}
int count = 1;
int m= 1;
for (auto p : hash) {
    if (hash.find(p.first + 1)==hash.end()) {
        count = 1;
    }
    else {
        count++;
    }
    m = max(m, count);
}
return m;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s=set(nums)
        ans=0
        for num in s:
            if num-1 not in s:
                count=1
                curr=num

                while curr+1 in s:
                    count+=1
                    curr+=1
                ans=max(ans,count)
        return ans
        